### ISSUE TITLE
Need to define the terms "known Unicode property name/alias/value"

### ISSUE THREAD

**Issue by allenwb** [tc39/proposal-regexp-unicode-property-escapes#15]
These terms are used throughout the spec. but not clearly defined. 

Is "known" intended to mean something like "supported by this implementation" or "required by this specification"? 

The definition of UnicodeMatchProperty types to define which  property names and alias "Implementations must support", but I find that definition hard to interpret. What does "available" mean in that context.  How is "remaining, but not limited to" supposed to be interpreted. Remaining from what? Can implementation defined their own names/values?

Some of the statements in the NOTES for UnicodeMatchProperty/UnicodeMatchPropertyValue seem like normative requirements and hence should not be labeled as informative.

Basically, the whole description of what properties/alias/values are required needs to be made clearer and more explicit.

**Comment by mathiasbynens** [comment]
Thanks! I’ll look into clarifying this entire section.

One question though:

> Some of the statements in the NOTES for UnicodeMatchProperty/UnicodeMatchPropertyValue seem like normative requirements and hence should not be labeled as informative.

Which statements specifically? AFAICT, everything in those notes is implied by the normative requirements above. The notes are just there for clarification.

**Comment by allenwb** [comment]
Here is a rough suggestion for how you might restructure the prose part of UnicodeMatchProperty:

Valid property names are: **Any**, **ASCII**, **Assigned**; all properties classified by UAX44  as being in categories **General_Category**, **Script**, or **Script_Extensions**; and, all properties classified by UAX44 as binary properties. The property alias provided by UAX44 for these property names are valid alternatives to these names. Only the canonical forms [ref??] of these property names and property aliases are valid. Implementations must not recognize as valid not any other enumeration properties.

NOTE 1: This set of valid property names is a superset of what UTS18 RL1.2 requires.
NOTE 2: **Any**, **ASCII**, and **Assigned** are defined by UTS18.
NOTE 3: For example, **Script_Extensions** and **scx** are valid canonical names, but **script_extensions** or **Scx** are not.
NOTE 4: Valid UAX44 property alias are listed in PropertyAliases.txt.
